File: CarPooling.cpp
class Solution {
public:
    bool carPooling(vector<vector<int>>& trips, int capacity) {
        int n = trips.size();
        vector<int> v(1001,0);
        for(int i = 0; i<n ; i++)
        {
            v[trips[i][2]] += trips[i][0];
            v[trips[i][1]] -= trips[i][0];
        }
        
        for(int i = v.size() -2 ; i >= 0; i--)
        {
            v[i] += v[i+1];
            if(v[i] > capacity)
                return false;
        }
        
        
        return true;
    }
};

File: CarPooling.py
class Solution:
    def carPooling(self, trips: List[List[int]], capacity: int) -> bool:
        position = [0] * 1001

        for trip in trips:
            position[trip[1]] += trip[0]
            position[trip[2]] -= trip[0]

        if position[0] > capacity:
            return False

        for i in range(1,len(position)):
            position[i] += position[i-1]

            if position[i] > capacity:
                return False

        return True
